src/models/AssetPrice.ts

```diff
@@ -50,12 +59,21 @@ export default class AssetRewards {
         return [...Array.from(this.rewards.keys())]
     }
 
-    public setReceiver(receiver: string, amount: BigNumber): AssetRewards {
+    public setReceiver(receiver: string, amount: BigNumber): AssetPrice {
         this.rewards.set(receiver, amount)
-        this.totalPrice = AssetRewards.sumAmounts(this.getAmounts())
+        this.totalPrice = AssetPrice.sumAmounts(this.getAmounts())
         return this
     }
 
+    public setTokenAddress(address: string): AssetPrice {
+        this.tokenAddress = address
+        return this
+    }
```

Comment:
should we allow the tokenAddress to be set outside the constructor?
even if we do I don't think there is a need to return anything here. it makes it confusing because we are updating the current instance not returning a new one